src/components/Dialog/index.tsx

```diff
@@ -0,0 +1,23 @@
+import React, { ReactNode } from "react"
+import { Modal } from "../Modal"
+import { DialogHeader } from "./DialogHeader"
+import { DialogBody } from "./DialogBody"
+import { DialogFooter } from "./DialogFooter"
+import "./Dialog.scss"
+
+type TDialogProps = React.ComponentProps<typeof Modal> &{
+  children: ReactNode
+}
```

Comment:
```suggestion
type TDialogProps = React.ComponentProps<typeof Modal> 
```
the children is already inside the Modal type